tests/hikari/hikari_test_helpers.py

```diff
@@ -149,21 +149,22 @@ def mock_class_namespace(
     if init is False:
         namespace["__init__"] = lambda _: None
 
-    if implement_abstract_methods:
-        for method_name in getattr(klass, "__abstractmethods__", ()):
+    if implement_abstract_methods and hasattr(klass, "__abstractmethods__"):
+        for method_name in klass.__abstractmethods__:
             if method_name in namespace:
                 continue
 
             attr = getattr(klass, method_name)
 
             if inspect.isdatadescriptor(attr) or inspect.isgetsetdescriptor(attr):
-                namespace[method_name] = mock.MagicMock(__isabstractmethod__=False)
+                namespace[method_name] = mock.PropertyMock(__isabstractmethod__=False)
```

Comment:
I used MagicMock on purpose at the time, I originally used propertymock and had issues with it. Can't remember what they were now so ig it is fine.

Reply:
This is might've been that trying to do obj.PropertyMock() = "foo" won't override the property mock which seems like sane behaviour to me